src/layout/stack.scss

```diff
@@ -0,0 +1,299 @@
+// Stack layout helper component
+
+.Stack {
+
+  // A Stack component lays elements horizontally or vertically on the page.
+  //
+  // Stack is a simple abstraction of CSS' Flexbox. Use it to structure elements
+  // that are visually grouped, following the same direction.
+  //
+  // Markup structure
+  // =========
+  //
+  // .Stack
+  // ├─ &.Stack--dir-[ inline | block ]-[ whenRegular | whenNarrow ]
+  // ├─ &.Stack--gap-[ none | condensed | normal | spacious ]-[ whenRegular | whenNarrow ]
+  // ├─ &.Stack--align-[ start | center | end | baseline ]-[ whenRegular | whenNarrow ]
+  // ├─ &.Stack--alignWrap-[ start | center | end | distribute | distributeEvenly ]-[ whenRegular | whenNarrow ]
+  // ├─ &.Stack--spread-[ start | center | end | distribute | distributeEvenly ]-[ whenRegular | whenNarrow ]
+  // ├─ &.Stack--wrap-[ whenRegular | whenNarrow ]
+  // ├─ &.Stack--nowrap-[ whenRegular | whenNarrow ]
+  // ├─ &.Stack--showDividers-[ whenRegular | whenNarrow ]
+  // │
+  // ├─ .Stack-divider
+
+  --Stack-gap: var(--primer-stack-gap-normal, 16px);
+  --Stack-gap-whenNarrow: var(--Stack-gap);
+  --Stack-divider-color: var(--color-border-default);
+
+  display: flex;
+  flex-flow: column;
+  flex-wrap: nowrap;
```

Comment:
> Should Stack also support something like `flex: 1` so that certain child elements take up the available space. Maybe with `Stack-item--flex`, `Stack-item--expanded` or similar. Or is it ok to keep using the flex-1 utility for that.

Yeah, I think it makes sense to have an optional `Stack-item` in the API so users can use `--expand` or `--keep-size` modifier classes. I'll update the PR with these considerations, thank you! 🙇 

> This is more for PVC/PRC... can `Stack-divider` also be used to group certain child elements together or it's more an either all or nothing.

I think it should be available in PVC/PRC to be consumed individually, too. c/c @mperrotti

Reply:
@simurai I've added `Stack-item` (and also cleaned a lot of the CSS redundancies) feel free to take a look again 🙇